### PR TITLE
Austin/image caching

### PR DESCRIPTION
**Pull request by AAAstorga**
Oh baby I made a ton of changes, sorry

- Added proper tab bar icons
- revamped filter range slider
- revamped Class list page
- added new delegates
- new fonts
- gave the app an uplift
- added brand new apply filter button
- cleaned up styling/code whenever I could
- added image caching
- added a bunch of other things I forget

*certified fresh*

### CODE REVIEW

File: Fitness/Networking/NetworkManager.swift [hunk 1]
```diff
@@ -187,4 +196,10 @@ struct NetworkManager {
             completion(Array(instructors))
         }
     }
+
+    // MARK: - Image Caching
+    private func cacheImage(imageUrl: URL, isClassCancelled: Bool = false) {
```
Comment (jhseung) [inline Fitness/Networking/NetworkManager.swift]: What is the `isClassCancelled` bool being used for? Can we remove?

File: Fitness/Views/Cells/AllGymsCell.swift [hunk 2]
```diff
@@ -3,7 +3,7 @@
 //  Fitness
 //
 //  Created by Joseph Fulgieri on 3/18/18.
-//  Copyright © 2018 Keivan Shahida. All rights reserved.
+//  Copyright © 2018 Uplift. All rights reserved.
```
Comment (jhseung) [inline Fitness/Views/Cells/AllGymsCell.swift]: <3

File: Fitness/Views/Headers/ClassListHeaderView.swift [hunk 3]
```diff
@@ -129,10 +134,7 @@ class ClassListHeaderView: UITableViewHeaderFooterView, UICollectionViewDataSour
         let cell = collectionView.cellForItem(at: indexPath) as! CalendarCell
 
         cell.dateLabel.textColor = .white
-        cell.shapeLayer.fillColor = UIColor.fitnessBlack.cgColor
-        delegate.selectedDate = cell.date
-
-        delegate.updateDate()
+        //cell.shapeLayer.fillColor = UIColor.fitnessBlack.cgColor
```
Comment (jhseung) [inline Fitness/Views/Headers/ClassListHeaderView.swift]: remove all the commented out code containing `shapeLayer`

File: Fitness/Controllers/GymDetailViewController.swift [hunk 4]
```diff
@@ -471,22 +475,22 @@ extension GymDetailViewController: UITableViewDataSource {
...
-
-            cell.classLabel.text = gymClassInstance.className
-            cell.timeLabel.text = Date.getStringDate(date: gymClassInstance.startTime)
-            cell.timeLabel.text = cell.timeLabel.text?.removeLeadingZero()
-
-            cell.instructorLabel.text = gymClassInstance.instructor
-
-            cell.duration = Int(gymClassInstance.duration) / 60
-            cell.durationLabel.text = String(cell.duration) + " min"
+//            let cell = tableView.dequeueReusableCell(withIdentifier: ClassListCell.identifier, for: indexPath) as! ClassListCell
```
[the remark this answers is not in the document]
Comment (AAAstorga) [reply Fitness/Controllers/GymDetailViewController.swift]: yes, sorry I didn't clarify

**Pull request merged**